sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -1327,6 +1318,21 @@ public Query visitReference(Reference symbol, Context context) {
             return super.visitReference(symbol, context);
         }
 
+        @Override
+        public Query visitLiteral(Literal literal, Context context) {
+            Object value = literal.value();
+            if (value == null) {
+                return Queries.newMatchNoDocsQuery("WHERE null -> no match");
+            }
+            try {
+                return (boolean) value
+                    ? Queries.newMatchAllQuery()
+                    : Queries.newMatchNoDocsQuery("WHERE false -> no match");
+            } catch (ClassCastException e) {
```

Comment:
When can this happen? can we add a comment?

Reply:
@matriv pushed a fixup